auth0-extensions/auth0-deploy-extensions
# [MFA-313] feat: add guardian policies, message types and selected provider

## ✏️ Changes
  
Add support for managing:
- MFA/Guardian Policies
- MFA/Guardian Phone Factor Message Types
- MFA/Guardian Phone Factor Selected Provider

Adds the `read:mfa_policies` and `update:mfa_policies` as required scope for the Auth0 Management API (needed to manage MFA Policies).

Bumps `auth0` to `v2.27.0` ([Diff](https://github.com/auth0/node-auth0/compare/v2.23.0...v2.27.0))
Bumps `auth0-source-control-extension-tools` to `v4.1.1` ([Diff](https://github.com/auth0-extensions/auth0-source-control-extension-tools/compare/v4.0.5...v4.1.1))

⚠️ Depends on:
- https://github.com/auth0-extensions/auth0-source-control-extension-tools/pull/96
- https://github.com/auth0/node-auth0/pull/501
  
## 📷 Screenshots

🚫 
  
## 🔗 References
  
https://auth0team.atlassian.net/browse/MFA-313
  
## 🎯 Testing
  
You need to:
- Setup the local dev environment (see the README)
  - Note: Need to get the fixes that are submitted at: https://github.com/auth0-extensions/auth0-deploy-extensions/pull/69
  - Create a repo in gitlab/github/bitbucket/visualstudio, and add some files:
    - ./guardian/policies.json
    - ./guardian/phoneFactorMessageTypes.json
    - ./guardian/phoneFactorSelectedProvider.json
   
🚫 This change has been tested in a Webtask

✅ This change has been tested locally

✅ This change has unit test coverage
  
🚫 This change has integration test coverage
  
🚫 This change has been tested for performance
  
## 🚀 Deployment
  

✅ This can be deployed any time

  
## 🎡 Rollout
  
In order to verify that the deployment was successful we will install the extension for a tenant and sync the MFA/Guardian settings.
  
## 🔥 Rollback
  
We will rollback if syncing MFA/Guardian settings fails or causes adverse effects.
  
### 📄 Procedure
  
Rollback the extension version.
 
## 🖥 Appliance
  
**Note to reviewers:** ensure that this change is compatible with the Appliance.
